_posts/2023-06-20-resume-parsing-insights-and-steps-to-create-your-own-parser.md

```diff
@@ -0,0 +1,900 @@
+---
+layout: post
+title: "Resume Parsing: Insights and Steps to Create Your Own Parser"
+tldr: "We will be using NLP techniques to extract information from resume."
+modified: 2023-06-20 13:40:00 +0530
+category: technology
+tags: [Resume-parser, Python, Flask, NLP, LLM]
+author: madhav
+image:
+  feature: 
+  credit: 
+  creditlink: 
+comments: 
+share: 
+---
+
+## What is Resume Parsing
+
+Resume parsing is the automated process of extracting relevant information from resumes or CVs. 
+It analyzes the unstructured text of a resume and extracts specific details like contact information, work experience, education, skills, and achievements. 
+The extracted data is then converted into a structured format, allowing for easy analysis and integration into recruitment systems.
+
+## Benefits of Resume Parsing
+
+- It is a time-saving automation
+- It increases efficiency in candidate screening
+- Improves accuracy in data extraction
+- It standardizes the data extraction and formatting
+- It is scalable for handling large volumes of resumes
+
+
+## What you'll learn from this blog:
+
+1. Resume parsing techniques for different file formats.
+2. Extracting specific details from resumes.
+3. Leveraging NLP techniques for parsing.
+4. Handling multicolumn resumes.
+5. Dockerizing the Application: Simplifying Deployment and Scalability
+6. Hosting it on AWS EC2.
+
+**Let's get Started 🎉**
+
+We'll utilize Python and its Flask framework to create a resume parsing server.
+
+## Application Flow Chart:
+
+![Application Flow Chart Image]( {{site.baseurl}}/images/resume-parsing-insights-and-steps-to-create-your-own-parser/file-flow.jpg)
+
+We will be primarily working on 3 categories of file formats:
+1. PDF
+2. DOCX
+3. Images (.png, .jpg, etc.)
+
+### Data that we will be extracting
+
+1. Embedded links in PDF
+2. Personal data: <br>
+    2.1. Name: First name and last name <br>
+    2.2. Email <br>
+    2.3. Phone Number <br>
+    2.4. Address: City, Country, and Zip code <br>
+    2.5. Links: Social and Coding Platform links <br>
+3. All other sections present in resume 
+4. Education <br>
+    4.1. Institute name <br>
+    4.2. Duration: Start date and End date <br>
+    4.3. Grade/CGPA <br>
+    4.4. Degree <br>
+5. Experience<br>
+    5.1. Company name<br>
+    5.2. Role<br>
+    5.3. Durations: Start date and End date<br>
+    5.4. Skills<br>
+6. Certification: <br>
+    6.1. Description <br>
+    6.2. Duration <br>
+    6.3. Skill <br>
+7. Project: <br>
+    7.1. Project name <br>
+    7.2. Skills <br>
+    7.3. Description <br>
+8. Skills
+9. Achievements
+10. Exam scores<br>
```

Comment:
I don't think we have exam scores extraction logic?

Reply:
I have added it.

Reply:
Can you point to the line where you have written this logic?

Reply:
in file `exams_extract.py` line 16